src/Components/Web.Extensions/src/HeadManagement/TitleElement.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal readonly struct TitleElement : IHeadElement
```

Comment:
We could have two separate method calls for head and the rest. I don't think you really need `IHeadElement`

Reply:
I added `IHeadElement` more so that we would could be confident in JS that the JSON included in a prerendered head comment would include the `type` property. Without the interface, it's possible for someone to call `BuildHeadElementComment<TElement>` where `TElement` doesn't contain a `Type` property. But yeah, this might be too much abstraction for something we could easily catch at runtime in the `default` case you were suggesting.